### PR TITLE
Add support for filtering images by tag

### PR DESCRIPTION
**Pull request by krijnr**

### CODE REVIEW

**Comment by krijnr** [comment]
[Issue 6: Images by Tag](https://trello.com/c/Uk5H1xPK/328-issue-6-images-by-tag)

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -71,4 +71,19 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
       end
     end
   end
+
+  test 'should list all images with specific tag only' do
+    Image.create! path: 'https://www.demo.com/image3.jpg', tag_list: 'tag_b,tag_d'
```
Comment (infiniteluke) [inline test/controllers/images_controller_test.rb:76]: `'tag_b,tag_d'` --> `'tag_b, tag_d'`

File: app/views/images/index.html.erb [hunk 2]
```diff
@@ -8,7 +8,7 @@
     <div class="row">
       <%= image_tag(image.path, class:"width-400 js-image-list") %>
       <% image.tags.each do |tag| %>
-        <span class="badge badge-secondary js-tag-list"><%= tag.name %></span>
+        <%= link_to tag.name, images_path(tag_filter:tag.name), class:"badge badge-secondary js-tag-list" %>
```
Comment (infiniteluke) [inline app/views/images/index.html.erb:11]: `class:"badge` --> `class: "badge`

File: test/controllers/images_controller_test.rb (hunk 1 above)
Comment (krijnr) [reply test/controllers/images_controller_test.rb]: I fixed the white space issue

File: app/views/images/index.html.erb (hunk 2 above)
Comment (krijnr) [reply app/views/images/index.html.erb]: I fixed the white space issue

**Pull request merged**